lib/rubygems.rb

```diff
@@ -941,7 +941,7 @@ def self.ui
   def self.use_paths(home, *paths)
     paths = nil if paths == [nil]
     paths = paths.first if Array === Array(paths).first
-    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths }
+    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths.join(Gem.path_separator) }
```

Comment:
this when fail when `Gem.path_separator` is a regex, as it is on Jruby @tenderlove (see https://github.com/rubygems/rubygems/pull/1469)